python_modules/dagster/dagster/_core/definitions/definitions_class.py

```diff
@@ -114,7 +113,48 @@ def created_repo():
 
 class Definitions:
     """
-    A set of definitions to be explicitly available and loadable by Dagster tools.
+    A set of definitions explicitly available and loadable by Dagster tools.
+
+    Parameters:
+        assets (Optional[Iterable[Union[AssetsDefinition, SourceAsset, CacheableAssetsDefinition]]]):
+            A list of assets for this repository. Assets can be created by annotating
+            a function with :py:func:`@asset <asset>` or
+            :py:func:`@observable_source_asset <observable_source_asset>`.
+            Or they can by directly instantiating :py:class:`AssetsDefinition`,
+            :py:class:`SourceAsset`, or :py:class:`CacheableAssetsDefinition`.
+
+        schedules (Optional[Iterable[ScheduleDefinition]]):
+            List of schedules.
+
+        sensors (Optional[Iterable[SensorDefinition]]):
+            List of sensors, typically created with :py:func:`@sensor <sensor>`.
+
+        jobs (Optional[Iterable[Union[JobDefinition, UnresolvedAssetJobDefinition]]]):
+            List of jobs. Typically created with :py:func:`define_asset_job <define_asset_job>`
+            or with :py:func:`@job <job>` for jobs defined in terms of ops directly.
+            Jobs created with :py:func:`@job <job>` must already have resources bound
+            at job creation time. They do not respect the `resources` argument here.
+
+        resources (Optional[Mapping[str, Any]]): Dictionary of resources to bind to assets.
+            The resources dictionary takes raw Python objects,
+            not just instances of :py:class:`ResourceDefinition`. If that raw object inherits from
+            :py:class:`IOManager`, it gets coerced to an :py:class:`IOManagerDefinition`.
+            Any other object is coerced to a :py:class:`ResourceDefinition`.
+            These resources will be automatically bound
+            to any assets passed to this Definitions instance using
+            :py:func:`with_resources <with_resources>`. Assets passed to Definitions with
+            resources already bound using :py:func:`with_resources <with_resources>` will
+            override this dictionary.
+
+        executor (Optional[ExecutorDefinition]):
+            Default executor for jobs. Individual jobs
```

Comment:
I think it's worth mentioning that this will be used even outside the context of jobs - i.e. whenever an asset is materialized. While we technically represent that a jobs under the covers, that's not how it's exposed to users.